andela/travel_tool_back
# #160047860 Create Approvals Model and Endpoint

#### What does this PR do?
- Creates an endpoint for fetching approvals
- Creates a database table for a Approvals
- Adds an association between Approval and Request models
- Enables filtering of approvals by status
- Enables pagination of responses from the approvals endpoint

#### Description of Task to be completed?
The task to be completed was to enable the filters for the approvals data table on frontend. This required an integration of the table to the backend to fetch true responses and replace the dummy approval placeholders on the frontend application.

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Change into the project directory -` cd travel_tool_back`
- Navigate to [travela](http://travela-staging.andela.com/) in your browser and login
- Open the developer tools, locate applications tab, copy `jwt-token` in the cookies
- Check `.env.example` and set up your `.env` appropriately
- Get back to your command line, Install dependencies and start the server - `make start`
- In`Postman`, use the `jwt-token` to send a request by adding a `Header`, key `Authorization` and value the `token`
- You should be able to get responses for `GET` requests to: 
  - `localhost:5000/api/v1/approvals`
  - `localhost:5000/api/v1/approvals?status=open`
  - `localhost:5000/api/v1/approvals?status=past`
  - You can also include pagination query parameters e.g. `page=3&limit-15`
  - These responses should include your approvals, pending and past, if there are any for you

#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?
#160047860

#### Screenshots (if appropriate)
<img width="624" alt="screen shot 2018-09-05 at 6 49 34 am" src="https://user-images.githubusercontent.com/23254512/45070003-f3d3c780-b0d7-11e8-84cb-0f92fefcaacb.png">

<img width="644" alt="screen shot 2018-09-05 at 7 17 28 am" src="https://user-images.githubusercontent.com/23254512/45070798-d4d73480-b0db-11e8-8ea9-6648101e1e2e.png">


#### Questions:
